eventstore/test_whatsapp_actions.py

```diff
@@ -239,8 +244,222 @@ def test_handle_whatsapp_hsm_error_unsent_disabled(self):
+        event.is_message_expired_error.return_value = False
 
         with patch("eventstore.tasks.rapidpro") as p:
             handle_whatsapp_hsm_error(event)
 
         p.create_flow_start.assert_not_called()
+
+    @responses.activate
+    @patch("eventstore.tasks.get_utc_now")
+    def test_handle_whatsapp_message_expired_error(self, mock_get_utc_now):
+        """
+        Sends a SMS and updates the contact if the contact hasn't been sent this
+        message in 30 days
+        """
+        timestamp = 1543999390.069308
+        mock_get_utc_now.return_value = datetime.datetime.fromtimestamp(timestamp)
+
+        event = Mock()
+        event.recipient_id = "27820001001"
+        event.is_hsm_error.return_value = False
+
+        tasks.rapidpro = TembaClient("textit.in", "test-token")
+
+        responses.add(
+            responses.GET,
+            "https://textit.in/api/v2/contacts.json?urn=whatsapp:27820001001",
+            json={
+                "results": [
+                    {
+                        "uuid": "contact-id",
+                        "name": "",
+                        "language": "zul",
+                        "groups": [],
+                        "fields": {},
+                        "blocked": False,
+                        "stopped": False,
+                        "created_on": "2015-11-11T08:30:24.922024+00:00",
+                        "modified_on": "2015-11-11T08:30:25.525936+00:00",
+                        "urns": ["tel:+27820001001"],
+                    }
+                ],
+                "next": None,
+            },
+        )
+
+        responses.add(responses.POST, "http://turn/v1/messages", json={})
+
+        responses.add(
+            responses.POST,
+            "https://textit.in/api/v2/contacts.json?urn=whatsapp:27820001001",
+            json={
+                "uuid": "contact-id",
+                "name": "",
+                "language": "zul",
+                "groups": [],
+                "fields": {},
+                "blocked": False,
+                "stopped": False,
+                "created_on": "2015-11-11T08:30:24.922024+00:00",
+                "modified_on": "2015-11-11T08:30:25.525936+00:00",
+                "urns": ["tel:+27820001001"],
+            },
+        )
+
+        handle_whatsapp_message_expired_error(event)
+
+        [_, turn_call, rapidpro_post] = responses.calls
+
+        self.assertEqual(
+            json.loads(turn_call.request.body),
+            {
+                "preview_url": False,
+                "recipient_type": "individual",
+                "to": "27820001001",
+                "type": "text",
+                "text": {
+                    "body": "Sibona ukuthi imiyalezo yeWhatsApp yakho yeMomConnect "
+                    "ayifikanga. Ukuthola imiyalezo nge-SMS, phendula uthi "
+                    "\u201cSMS\u201d (kusebenza izindleko ezijwayelekile)."
```

Comment:
These probably shouldn't be curly quotes, but that's a translation file issue, not a code issue here.